backend/src/main/java/com/woowacourse/levellog/prequestion/application/PreQuestionService.java

```diff
@@ -0,0 +1,88 @@
+package com.woowacourse.levellog.prequestion.application;
+
+import com.woowacourse.levellog.common.exception.InvalidFieldException;
+import com.woowacourse.levellog.levellog.domain.Levellog;
+import com.woowacourse.levellog.levellog.domain.LevellogRepository;
+import com.woowacourse.levellog.levellog.exception.LevellogNotFoundException;
+import com.woowacourse.levellog.member.domain.Member;
+import com.woowacourse.levellog.member.domain.MemberRepository;
+import com.woowacourse.levellog.member.exception.MemberNotFoundException;
+import com.woowacourse.levellog.prequestion.domain.PreQuestion;
+import com.woowacourse.levellog.prequestion.domain.PreQuestionRepository;
+import com.woowacourse.levellog.prequestion.dto.PreQuestionDto;
+import com.woowacourse.levellog.prequestion.exception.PreQuestionNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class PreQuestionService {
+
+    private final PreQuestionRepository preQuestionRepository;
+    private final LevellogRepository levellogRepository;
+    private final MemberRepository memberRepository;
+
+    @Transactional
+    public Long save(final PreQuestionDto request, final Long levellogId, final Long memberId) {
+        final Levellog levellog = getLevellog(levellogId);
+        final Member from = getMember(memberId);
+
+        return preQuestionRepository.save(request.toEntity(levellog, from))
+                .getId();
+    }
+
+    public PreQuestionDto findById(final Long preQuestionId, final Long levellogId) {
+        final PreQuestion preQuestion = getPreQuestion(preQuestionId);
+        validateLevellog(preQuestion, levellogId);
+
+        return PreQuestionDto.from(preQuestion);
+    }
+
+    @Transactional
+    public void update(final PreQuestionDto request, final Long preQuestionId, final Long levellogId, final Long memberId) {
+        final PreQuestion preQuestion = getPreQuestionByFromMember(preQuestionId, memberId);
+        validateLevellog(preQuestion, levellogId);
+
+        preQuestion.update(request.getPreQuestion());
+    }
```

Comment:
```suggestion
    public void update(final PreQuestionDto request, final Long preQuestionId, final Long levellogId, final Long memberId) {
        final PreQuestion preQuestion = getPreQuestion(preQuestionId);
        
        // preQuestion의 from이 member가 아니면 예외를 던지는 로직
        
        validateLevellog(preQuestion, levellogId);

        preQuestion.update(request.getPreQuestion());
    }
```
이런식으로 예외 던지는 상황을 쪼개는건 어떨까요?
지금은 `PreQuestion` PK에 해당하는 값이 존재해도 작성자가 아닌경우라면 `PreQuestionNotFoundException`이 터지네요

Reply:
조금 더 정확하고 구체적으로 예외를 잡을 수 있도록 수정했습니다. 가능하면 메서드 하나가 하나의 일만 하도록 수정하면서 수정해봤습니다.